Src/GoogleApis.Auth/OAuth2/ComputeCredential.cs

```diff
@@ -36,6 +38,15 @@ namespace Google.Apis.Auth.OAuth2
     /// </summary>
     public class ComputeCredential : ServiceCredential
     {
+        /// <summary>The metadata server url.</summary>
+        public const string MetadataServerUrl = "http://metadata.google.internal";
+
+        /// <summary>
+        /// Experimentally, 200ms was found to be 99.9999% reliable. 
+        /// This is a conservative timeout to minimize hanging on some troublesome network. 
+        /// </summary>
+        private const int MetadataServerPingTimeoutInMilliseconds = 1000;

```

Comment:
We had talked to compute engine team around this and it was decided that we don't need a retry as this is meant to be 100% reliable on GCE (their words not mine, they indicated we will have bigger problems if this was anything less than 100% as it was on the nanny and not an off-box call). For off-GCE scenarios, we don't want to retry as we will end up in a pathological wait. We should fail fast and let the caller handle it.